src/main/java/hudson/plugins/analysis/core/PluginDescriptor.java

```diff
@@ -122,21 +128,6 @@ public PluginDescriptor(final Class<? extends Publisher> clazz) {
     }
 
     @Override
-    public Publisher newInstance(final StaplerRequest req, final JSONObject formData)
-            throws hudson.model.Descriptor.FormException {
-        JSONObject converted;
-        if (formData.containsKey(CONFIGURATION_SECTION_KEY)) {
-            JSONObject old = formData.getJSONObject(CONFIGURATION_SECTION_KEY);
-            formData.put(CONFIGURATION_SECTION_KEY, convertHierarchicalFormData(old));
-            converted = formData;
-        }
-        else {
-            converted = convertHierarchicalFormData(formData);
-        }
-        return super.newInstance(req, converted);
-    }
```

Comment:
Yes. It was added because the JSON coming from the UI was something like this:

```
...
"pattern": "",
"isRankActivated": false,
"includePattern": "",
"excludePattern": "",
"canRunOnFailed": false,
"shouldDetectModules": false,
"healthy": "",
"unHealthy": "",
"thresholdLimit": "low",
"unstableTotalAll": "",
"unstableTotalHigh": "",
"unstableTotalNormal": "",
"unstableTotalLow": "",
"failedTotalAll": "",
"failedTotalHigh": "",
"failedTotalNormal": "",
"failedTotalLow": "",
"canComputeNew": {
    "unstableNewAll": "10",
    "unstableNewHigh": "",
    "unstableNewNormal": "",
    "unstableNewLow": "",
    "failedNewAll": "",
    "failedNewHigh": "",
    "failedNewNormal": "",
    "failedNewLow": "",
    "useDeltaValues": true,
    "usePreviousBuildAsReference": false,
    "useStableBuildAsReference": false
},
"defaultEncoding": ""
...
```
The ```newInstace``` override was basically flattening the ```canComputeNew``` field in order to make the JSON matching the Java model and let stapler do the rest. This process has been replaced by using the ```inline="true"``` property in ```thresholds.jelly``` (```<f:optionalBlock name="canComputeNew" inline="true"```) which makes exactly what the newInstance override was for.

The change is tested and working for both Freestyle jobs, Maven jobs and Workflow jobs.